stdlib/SparseArrays/src/linalg.jl

```diff
@@ -1,12 +1,35 @@
 # This file is a part of Julia. License is MIT: https://julialang.org/license
 
-import LinearAlgebra: checksquare
+import LinearAlgebra: checksquare, sym_uplo
 using Random: rand!
 
 # In matrix-vector multiplication, the correct orientation of the vector is assumed.
 const StridedOrTriangularMatrix{T} = Union{StridedMatrix{T}, LowerTriangular{T}, UnitLowerTriangular{T}, UpperTriangular{T}, UnitUpperTriangular{T}}
 const AdjOrTransStridedOrTriangularMatrix{T} = Union{StridedOrTriangularMatrix{T},Adjoint{<:Any,<:StridedOrTriangularMatrix{T}},Transpose{<:Any,<:StridedOrTriangularMatrix{T}}}
 
+for (Wrapper, operation) ∈ [(:Hermitian, :adjoint), (:Symmetric, :transpose)]
+    @eval begin
+        (+)(A::SparseMatrixCSC, B::$Wrapper{<:Any,<:AbstractSparseMatrix}) = A + sparse(B)
+        (+)(A::$Wrapper{<:Any,<:AbstractSparseMatrix}, B::SparseMatrixCSC) = B + A
+
+        (+)(A::SparseMatrixCSC, B::$Wrapper) = A + collect(B)
+        (+)(A::$Wrapper, B::SparseMatrixCSC) = B + A
+
+        function +(A::$Wrapper{<:Any, <:SparseMatrixCSC}, B::$Wrapper{<:Any, <:SparseMatrixCSC})
+            if A.uplo == B.uplo
+                $Wrapper(parent(A) + parent(B), sym_uplo(A.uplo))
+            elseif A.uplo == 'U'
+                $Wrapper(parent(A) + $operation(parent(B)), :U)
+            else
+                $Wrapper($operation(parent(A)) + parent(B), :U)
+            end
+        end
+    end
+end
+(+)(A::Symmetric{<:Any, <:SparseMatrixCSC}, B::Hermitian{<:Any, <:SparseMatrixCSC}) = sparse(A) + B
```

Comment:
I'm not sure, but this boils down to
```suggestion
(+)(A::Symmetric{<:Any, <:SparseMatrixCSC}, B::Hermitian{<:Any, <:SparseMatrixCSC}) = sparse(A) + sparse(B)
```
anyway, so we might as well write it out here, given one-liner. Same below.